ocs_ci/ocs/cnv/virtual_machine.py

```diff
@@ -567,7 +614,160 @@ def delete(self):
         if self.volume_interface == constants.VM_VOLUME_PVC:
             self.pvc_obj.delete()
             self.volumeimportsource_obj.delete()
-        if self.volume_interface == constants.VM_VOLUME_DV:
+        elif self.volume_interface == constants.VM_VOLUME_DV:
+            self.dv_obj.delete()
+        if self.ns_obj:
+            self.ns_obj.delete_project(project_name=self.namespace)
+
+
+class VMCloner(VirtualMachine):
+    """
+    Class for handling cloning of a Virtual Machine.
+    Inherits from VirtualMachine to have access to its attributes and methods.
+    """
+
+    def __init__(self, vm_name, namespace=None):
+        """
+        Initializes cloned vm obj
+        """
+        super().__init__(vm_name=vm_name, namespace=namespace)
+        self.source_vm_obj = None
+        self.source_pvc_name = ""
+        self.dv_cr_data_obj = self.dv_rb_data_obj = None
+
+    def clone_vm(self, source_vm_obj, volume_interface, ssh=True, verify=True):
+        """
+        Clone an existing virtual machine.
+
+        Args:
+            source_vm_obj (VirtualMachine): The source VM object to clone.
+            volume_interface (str): The volume interface to use.
+            ssh (bool): Whether to verify SSH connectivity.
+            verify (bool): Whether to verify the VM status after cloning
+        """
+        self.source_vm_obj = source_vm_obj
+        self.source_pvc_name = source_vm_obj.pvc_name
+        self.source_ns = source_vm_obj.namespace
+        self.volume_interface = source_vm_obj.volume_interface
+        self.sc_name = source_vm_obj.sc_name
+        self.pvc_size = source_vm_obj.pvc_size
+        self.pvc_access_mode = source_vm_obj.pvc_access_mode
+
+        # Use methods from the parent class
+        self._create_namespace_if_not_exists()
+        vm_data = self._prepare_vm_data()
+        if ssh:
+            self._add_ssh_key_to_vm(vm_data)
+        # Handle cloning based on volume interface
+        if volume_interface == constants.VM_VOLUME_PVC:
+            self._clone_vm_pvc(vm_data=vm_data)
+        elif volume_interface == constants.VM_VOLUME_DV:
+            self._clone_vm_data_volume(vm_data=vm_data)
+        elif volume_interface == constants.VM_VOLUME_DVT:
+            self._configure_dvt_clone(vm_data=vm_data)
+
+        vm_ocs_obj = create_resource(**vm_data)
+        logger.info(f"Successfully cloned VM: {vm_ocs_obj.name}")
+
+        if verify:
+            self.verify_vm(verify_ssh=True)
+
+    def _clone_vm_pvc(self, vm_data):
```

Comment:
Need to address encrypted pvcs as well.

Reply:
What specifically needs to be handled here?, since encryption steps are done in the multi_cnv fixture you have created. Just pass the storageclass which has encryption configured.